WEBEU3 - Temitope akinsoto - WEEK13 DAY 05 SPRINT CHALLENGE

temitopeakinsoto opened the pull request.

temitopeakinsoto committed:
solutions to self-study questions

temitopeakinsoto committed:
set up express api server

temitopeakinsoto committed:
install npm dependencies and create .env file

temitopeakinsoto committed:
set up server.js file for implementing actionsRouter and projectsRouter

temitopeakinsoto committed:
create actionsRouter and projectsRouter for handling CRUD requests to action and projects resources respectively

temitopeakinsoto committed:
fix bug while importing projectRouter and actionRouter files

temitopeakinsoto committed:
Add an endpoint for retrieving actions

In this commit, I created a new endpoint for retrieving the list of all actions from the database.

temitopeakinsoto committed:
Add an endpoint for retrieving projects

In this commit, I created a new endpoint for retrieving the list of all Projects from the database.

temitopeakinsoto committed:
Add a post endpoint to projects resource

In this commit, I added a post route which posts a new project to the project database.
A validate post middleware was also implemented to check if the new post to be inserted meets the requirements for a new project post

temitopeakinsoto committed:
Create a delete endpoint for projects resource

In this commit, I created a new endpoint for deleting a specified project by Id. This endpoint makes use of a validate if specified project exists or not

temitopeakinsoto committed:
Create a get endpoint for a specified project resource

In this commit, I created a new endpoint for retrieving a specified project by Id. This endpoint makes use of a validate if specified project exists or not

temitopeakinsoto committed:
Create a get endpoint for a specified ACTION resource

In this commit, I created a new endpoint for retrieving a specified action by Id. This endpoint makes use of a validateActionId middleware to check if the specified action exists or not

temitopeakinsoto committed:
MVP complete

temitopeakinsoto committed:
solution to self study

temitopeakinsoto committed:
MVP complete

twumm reviewed (approved):
Temi! 
I'm sure you are super proud of yourself with your first back-end week at Lambda.
You have done a good job so far and I will like to see you keep this pace up!
🚀

temitopeakinsoto committed:
REFACTOR: modify GET /actions/:id route

In this commit, I refactored the /actions/:id endpoint. Instead of making another database call to the actions table, I simply returned the req.actions object passed down from the validateActionId middleware

temitopeakinsoto committed:
Remove all unwanted console.log statements
